sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -240,6 +240,7 @@ class Analyzer(
       ExtractWindowExpressions ::
       GlobalAggregates ::
       ResolveAggregateFunctions ::
+      UseDecimalSum ::
```

Comment:
I think its ok to apply this rule after the resolution batch.

Reply:
Thanks @maropu for your comments. Based on the discussion here https://github.com/apache/spark/pull/27627#issuecomment-611308104, the changes are now back only in Sum, which removes the necessity of this rule.